### PR TITLE
Add schema validation hints to YAML files

### PR DESCRIPTION
**Pull request by uklotzde**
This allows to validate the file structure in VSCode while editing.

Adding the `check-jsonschema` pre-commit hook does not work yet, because validation of build.yml fails. Please see the comment I added in this file. Looks like the schema is wrong.

### CODE REVIEW

**Comment by Holzhaus** [comment]
> Adding the check-jsonschema pre-commit hook does not work yet, because validation of build.yml fails. Please see the comment I added in this file. Looks like the schema is wrong.

Hmm, to me the comment sounds like it works with the changes you made. Does it? If so, can you add the hook?

**Comment by uklotzde** [comment]
Oh, that

> > Adding the check-jsonschema pre-commit hook does not work yet, because validation of build.yml fails. Please see the comment I added in this file. Looks like the schema is wrong.
> 
> Hmm, to me the comment sounds like it works with the changes you made. Does it? If so, can you add the hook?

Oh, that change slipped through unintentionally. I don't even know if this works and this would be an ugly hack that differs from what is written in the GitHub docs.

**Comment by uklotzde** [comment]
Reverted. The schema error for this line:

```
String does not match the pattern of "^\$\{\{.*\}\}$".yaml-schema: https://json.schemastore.org/github-workflow
```

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -405,6 +407,8 @@ jobs:
     - name: "[macOS/Windows] Upload build to downloads.mixxx.org"
       # skip deploying Ubuntu builds to downloads.mixxx.org because these are deployed to the PPA
       if: runner.os != 'Linux' && github.event_name == 'push' && env.SSH_AUTH_SOCK != null
+      # FIXME: According to the schema expression syntax ${{ }} is required for the
+      # argument of `shell:`` in contradiction to the official GitHub docs!?
       shell: bash --login -eo pipefail "{0}"
```
Comment (Swiftb0y) [inline .github/workflows/build.yml:412]: The schema description seems overly strict.
```JSON
"shell": {
  "$comment": "https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#using-a-specific-shell",
  "description": "You can override the default shell settings in the runner's operating system using the shell keyword. You can use built-in shell keywords, or you can define a custom set of shell options.",
  "type": "string",
  "anyOf": [
    {
      "$ref": "#/definitions/expressionSyntax"
},
    {
      "$comment": "https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#custom-shell",
      "enum": [
        "bash",
        "pwsh",
        "python",
        "sh",
        "cmd",
        "powershell"
      ]
    }
  ]
},
```

Comment (Swiftb0y) [reply .github/workflows/build.yml:412]: The `${{ }}` seems to come from the `expressionSyntax` definition.

Comment (Holzhaus) [reply .github/workflows/build.yml:412]: I opened a PR at https://github.com/SchemaStore/schemastore/pull/2076.

**Comment by Holzhaus** [comment]
> Reverted. The schema error for this line:
> 
> ```
> String does not match the pattern of "^\$\{\{.*\}\}$".yaml-schema: https://json.schemastore.org/github-workflow
> ```

The schema was updated, please try again.

**Comment by uklotzde** [comment]
I also updated some other pre-commit dependencies in a finishing commit.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -94,13 +94,19 @@ repos:
...
 - repo: https://github.com/DavidAnson/markdownlint-cli2
   rev: v0.4.0
   hooks:
   - id: markdownlint-cli2
+- repo: https://github.com/sirosen/check-jsonschema
+  rev: 0.10.2
+  hooks:
+  - id: check-github-actions
+  # TODO: Checking .github/workflows/build.yml fails
+  #- id: check-github-workflows
```
Comment (Swiftb0y) [inline .pre-commit-config.yaml:109]: I don't really understand the purpose of this so far. We don't have any github actions and and the workflows are not being checked.

**Review by Swiftb0y — APPROVED** [review]
LGTM thanks

**Pull request merged**